mParticle-Responsys/MPKitResponsys.m

```diff
@@ -0,0 +1,294 @@
+#import <UserNotifications/UNUserNotificationCenter.h>
+#endif
+
+NSString * const PIOConfigurationAPIKey = @"apiKey";
+NSString * const PIOConfigurationAccountToken = @"accountToken";
+NSString * const PIOConfigurationConversionURL = @"conversionUrl";
+NSString * const PIOConfigurationRIAppID = @"riAppId";
+
+NSString * const CUSTOM_FLAG_IAM = @"Responsys.Custom.iam";
+NSString * const CUSTOM_FLAG_ENGAGEMENT = @"Responsys.Custom.e";
+
+@interface MPKitResponsys(){
+    PushIOManager *_pioManager;
+}
+
+@end
+
+@implementation MPKitResponsys
+
++(void) load{
+    MPKitRegister *pioKitRegister = [[MPKitRegister alloc] initWithName:@"Oracle Responsys" className: @"MPKitResponsys"];
+    [MParticle registerExtension:pioKitRegister];
+}
+
+- (instancetype)initWithConfiguration:(nonnull NSDictionary *)configuration startImmediately:(BOOL)startImmediately{
+    self = [super init];
+    NSString *apiKey = nil;
+    NSString *accountToken = nil;
+    NSString *conversionURL = nil;
+    NSString *riAppID = nil;
+
+    if (self) {
+        apiKey = configuration[PIOConfigurationAPIKey];
+        accountToken = configuration[PIOConfigurationAccountToken];
+        conversionURL = configuration[PIOConfigurationConversionURL];
+        riAppID = configuration[PIOConfigurationRIAppID];
+    }
+    
+    if ((NSNull *)apiKey == [NSNull null] || (NSNull *)accountToken == [NSNull null] || (NSNull *)conversionURL == [NSNull null] || (NSNull *)riAppID == [NSNull null])
+    {
+        return nil;
+    } else {
+        return self;
+    }
+}
+
+- (MPKitExecStatus *)didFinishLaunchingWithConfiguration:(NSDictionary *)configuration {
+    MPKitExecStatus *execStatus = nil;
+    self.configuration = configuration;
+    [self start];
+    execStatus = [[MPKitExecStatus alloc] initWithSDKCode:[MPKitResponsys kitCode] returnCode:MPKitReturnCodeSuccess];
+    [self start];
+    return execStatus;
+}
+
+-(void) start{
+    static dispatch_once_t kitPredicate;
+    
+    dispatch_once(&kitPredicate, ^{
+        self->_started = YES;
+        NSString *apiKey = self.configuration[PIOConfigurationAPIKey];
+        NSString *accountToken = self.configuration[PIOConfigurationAccountToken];
+        NSString *coversionURLString = self.configuration[PIOConfigurationConversionURL];
+        NSString *riAppID = self.configuration[PIOConfigurationRIAppID];
+        NSError *error = nil;
+        BOOL configured = [[self pushIOManager] configureWithAPIKey:apiKey accountToken:accountToken error:&error];
+        if (configured) {
+            [[self pushIOManager] setConversionURL:[NSURL URLWithString:coversionURLString]];
+            [[self pushIOManager] setRIAppID: riAppID];
+            [[self pushIOManager] setLogLevel:(([MParticle sharedInstance].environment == MPEnvironmentDevelopment) ? PIOLogLevelVerbose : PIOLogLevelError)];
+            [[self pushIOManager] registerForAllRemoteNotificationTypes:^(NSError *error, NSString *response) {
```

Comment:
Can you please avoid registering for push notifications here? Our customers require having control over when / how the push registration prompt is displayed to the user.